File: src/UE4_Packages/FSoftObjectPath.java
/**
 * 
 */
package UE4_Packages;

/**
 * @author FunGames
 *
 */
public class FSoftObjectPath {
	private String assetPathName;
	private String subPathString;
	
	public FSoftObjectPath(FArchive Ar, NameMap nameMap) throws ReadException {
		assetPathName = Ar.readFName(nameMap);
		subPathString = Ar.readString();
	}

	public String getAssetPathName() {
		return assetPathName;
	}

	public String getSubPathString() {
		return subPathString;
	}
}

File: src/main/kotlin/me/fungames/jfortniteparse/ue4/assets/objects/FSoftObjectPath.kt
package me.fungames.jfortniteparse.ue4.assets.objects

import me.fungames.jfortniteparse.ue4.UClass
import me.fungames.jfortniteparse.ue4.assets.reader.FAssetArchive
import me.fungames.jfortniteparse.ue4.assets.util.FName
import me.fungames.jfortniteparse.ue4.assets.writer.FAssetArchiveWriter

@Suppress("EXPERIMENTAL_API_USAGE")
class FSoftObjectPath : UClass {
    var assetPathName: FName
    var subPathString: String

    constructor(Ar: FAssetArchive) {
        super.init(Ar)
        assetPathName = Ar.readFName()
        subPathString = Ar.readString()
        super.complete(Ar)
    }

    fun serialize(Ar: FAssetArchiveWriter) {
        super.initWrite(Ar)
        Ar.writeFName(assetPathName)
        Ar.writeString(subPathString)
        super.completeWrite(Ar)
    }

    constructor(assetPathName: FName, subPathString: String) {
        this.assetPathName = assetPathName
        this.subPathString = subPathString
    }

    override fun toString() = "$assetPathName$subPathString"
}